src/task_queue/database_tasks/translate_contents.py

```diff
@@ -0,0 +1,176 @@
+                field_value = getattr(instance, field)
+                field_type = instance._meta.get_field(field)
+                
+                if isinstance(field_type, JSONField):
+                    if isinstance(field_value, dict):
+                        for key, value in field_value.items():
+                            self.translate_field(value, SOURCE_LANGUAGE_CODE)
+                else:
+                    self.translate_field(field_value, SOURCE_LANGUAGE_CODE)
+            return True
+        except Exception as e:
+            logger.log(f"Error occurred while translating model instance: {e}")
+            return False
+    
+    def translate_menu_model(self, model):
+        try:
+            instances = model.objects.all()
+            exclude_fields = ["link", "is_published", "id", "is_link_external"]
+            for instance in instances:
+                
+                content = JsonTranslator(instance.content, exclude_fields).get_flattened_dict()
+                footer_content = JsonTranslator(instance.footer_content, exclude_fields).get_flattened_dict()
+                
+                logger.log(f"Translating Nav contents for menu: {instance.name}")
+                for key, value in content.items():
+                    self.translate_field(value, SOURCE_LANGUAGE_CODE)
+                
+                logger.log(f"Translating Footer contents for menu: {instance.name}")
+                for key, value in footer_content.items():
+                    self.translate_field(value, SOURCE_LANGUAGE_CODE)
+                    
+            return True
+        except Exception as e:
+            logger.log(f"Error occurred while translating menu model: {e}")
+            return None
+        
+    def get_valid_instances(self, model):
+        query_params = {}
+        if hasattr(model, 'is_deleted'):
+            query_params['is_deleted'] = False
+        if hasattr(model, 'is_archived'):
+            query_params['is_archived'] = False
+        if hasattr(model, 'is_published'):
+            query_params['is_published'] = True
+        if hasattr(model, 'is_active'):
+            query_params['is_active'] = True
+        return model.objects.filter(**query_params)
+    
+    def load_db_contents_and_translate(self):
+        models = apps.get_models()
+        for model in models:
+            try:
+                # Process Menu model separately
+                if model.__name__ == "Menu":
+                    self.translate_menu_model(model)
+                    continue
+                
+                if not (hasattr(model, 'TranslationMeta') and model.TranslationMeta.fields_to_translate):
+                    continue
+                
+                logger.info(f"\n\nTranslating contents for model: {model.__name__}\n\n")
+                
+                instances = self.get_valid_instances(model)
+                
+                for instance in instances:
```

Comment:
why not serialize then call json translator instead?

Reply:
The serialized version of the instances includes a lot of unnecessary data due to the joins defined in the model. This approach gives us only the data of the model being processed.